core/arch/arm/kernel/thread_spmc.c

```diff
@@ -326,46 +327,107 @@ static bool is_my_uuid(uint32_t w0, uint32_t w1, uint32_t w2, uint32_t w3)
 	       w2 == 0x02005ebc && w3 == 0x1bc5d5a5;
 }
 
-static void handle_partition_info_get(struct thread_smc_args *args,
-				      struct ffa_rxtx *rxtx)
+void spmc_fill_partition_entry(struct ffa_partition_info *fpi,
+			       uint16_t endpoint_id, uint16_t execution_context)
 {
-	uint32_t ret_fid = 0;
-	int rc = 0;
+	fpi->id = endpoint_id;
+	/* Number of execution contexts implemented by this partition */
+	fpi->execution_context = execution_context;
 
-	if (!is_nil_uuid(args->a1, args->a2, args->a3, args->a4) &&
-	    !is_my_uuid(args->a1, args->a2, args->a3, args->a4)) {
-		ret_fid = FFA_ERROR;
-		rc = FFA_INVALID_PARAMETERS;
-		goto out;
+	/*
+	 * BIT(0): Supports receipt of direct requests
+	 * BIT(1): Can send direct requests
+	 * BIT(2): Cannot send and receive indirect messages
+	 * BIT(3): Does not support receipt of notifications
+	 * BIT(4-5): Partition ID is a PE endpoint ID
+	 */
+	fpi->partition_properties = FFA_PARTITION_DIRECT_REQ_RECV_SUPPORT |
+				    FFA_PARTITION_DIRECT_REQ_RECV_SUPPORT;
+}
+
+static uint32_t handle_partition_info_get_all(size_t *elem_count,
+					      struct ffa_rxtx *rxtx)
+{
+	struct ffa_partition_info *fpi = rxtx->tx;
+
+	/* Add OP-TEE SP */
+	spmc_fill_partition_entry(fpi, my_endpoint_id, 0);
```

Comment:
`s/0/CFG_TEE_CORE_NB_CORE/` if I got it right.

Reply:
Done